api/v1alpha1/promotion_types.go

```diff
@@ -7,12 +7,34 @@ import (
 type PromotionPhase string
 
 const (
-	PromotionPhasePending    PromotionPhase = "Pending"
-	PromotionPhaseInProgress PromotionPhase = "Promoting"
-	PromotionPhaseComplete   PromotionPhase = "Completed"
-	PromotionPhaseFailed     PromotionPhase = "Failed"
+	// PromotionPhasePending denotes a Promotion that has not been executed yet.
+	// i.e. It is currently waiting in a queue. Queues are stage-specific and
+	// prioritized by Promotion creation time.
+	PromotionPhasePending PromotionPhase = "Pending"
+	// PromotionPhaseRunning denotes a Promotion that is actively being executed.
+	//
+	// TODO: "Active" is the operative word here. We are leaving room for the
+	// possibility in the near future that an in-progress Promotion might be
+	// paused/suspended pending some user action.
+	PromotionPhaseRunning PromotionPhase = "Running"
+	// PromotionPhaseSucceeded denotes a Promotion that has been successfully
+	// executed.
+	PromotionPhaseSucceeded PromotionPhase = "Succeeded"
+	// PromotionPhaseErrored denotes a Promotion that has failed for technical
+	// reasons. Further information about the failure can be found in the
+	// Promotion's status.
+	//
+	// TODO: "For technical reasons" is the operative phrase here. We are leaving
+	// room for the possibility in the near future that a Promotion might fail
+	// as a result of some user action.
+	PromotionPhaseErrored PromotionPhase = "Errored"
 )
 
+// IsTerminal returns true if the PromotionPhase is a terminal one.
+func (p *PromotionPhase) IsTerminal() bool {
+	return *p == PromotionPhaseSucceeded || *p == PromotionPhaseErrored
```

Comment:
I  originally suggested using "complete" terminology, but I can see the argument that complete might be confused with success. So I'm okay with this.

Reply:
Oh... I knew you wanted to change what _was_ Completed to Succeeded...

I think I got my wires crossed a bit and thought we were getting rid of "Completed" entirely. Re-reading #600 now, I see you meant for "completed" to becomes what I now call "terminal."

Sorry for the confusion.

But as you noted, "terminal" may avoid further confusion, so in the words of the late, great Bob Ross: "We don't make mistakes. Just happy little accidents." :smile: